soc/arm/nordic_nrf/soc_nrf_common.h

```diff
@@ -15,6 +15,82 @@
 #include <devicetree.h>
 #include <toolchain.h>
 
+/**
+ * @brief Get a PSEL value out of a foo-gpios or foo-pin devicetree property
+ *
+ * Many Nordic bindings have 'foo-pin' properties to specify a pin
+ * configuration as a PSEL value directly instead of using a 'foo-gpios'
+ * <&gpioX Y flags> style controller phandle + GPIO specifier.
+ *
+ * It would be better to use 'foo-gpios' properties instead. This type
+ * of property is more in line with the recommended DT encoding for GPIOs.
+ *
+ * To allow for a smooth migration from 'foo-pin' to 'foo-gpios', this
+ * helper macro can be used to get a PSEL value out of the devicetree
+ * using whichever one of 'foo-gpios' or 'foo-pin' is in the DTS.
+ *
+ * Note that you can also use:
+ *
+ *   - NRF_DT_CHECK_PSEL_*() to check the property configuration at build time
+ *   - NRF_DT_GPIOS_TO_PSEL() if you only have a 'foo-gpios'
+ *
+ * @param node_id node identifier
+ * @param psel_prop lowercase-and-underscores old-style 'foo-pin' property
+ * @param gpios_prop new-style 'foo-gpios' property
+ * @param default_val the value returned if neither is set
+ * @return PSEL register value taken from psel_prop or gpios_prop, whichever
+ *         is present in the DTS. If gpios_prop is present, it is converted
+ *         to a PSEL register value first.
+ */
+#define NRF_DT_PSEL(node_id, psel_prop, gpios_prop, default_val)	\
+	COND_CODE_1(DT_NODE_HAS_PROP(node_id, psel_prop),		\
+		    (DT_PROP(node_id, psel_prop)),			\
+		    (COND_CODE_1(					\
+			    DT_NODE_HAS_PROP(node_id, gpios_prop),	\
+			    (NRF_DT_GPIOS_TO_PSEL(node_id,		\
+						  gpios_prop)),		\
+			    (default_val))))
+
+/**
+ * Error out the build if the devicetree node with identifier
+ * 'node_id' has both a legacy psel-style property and a gpios
+ * property.
+ *
+ * Otherwise, do nothing.
+ *
+ * @param node_id node identifier
+ * @param psel_prop lowercase-and-underscores PSEL style property
+ * @param psel_prop_name human-readable string name of psel_prop
+ * @param gpios_prop lowercase-and-underscores foo-gpios style property
+ * @param gpio_prop_name human-readable string name of gpios_prop
+ */
+#define NRF_DT_CHECK_PSEL_NOT_BOTH(node_id, psel_prop, psel_prop_name,	\
```

Comment:
Only noticed this in #32653:

Would `NRF_DT_PSEL_CHECK_NOT_BOTH` be a viable alternative?  It would make it easier to find API related to `DT_PSEL`.

Unless this is private API it might be too late, since `NRF_DT_CHECK_GPIO_CTLR*` landed before 2.5.

Reply:
> Unless this is private API it might be too late, since `NRF_DT_CHECK_GPIO_CTLR*` landed before 2.5.

This is a private API. I'll change it.